client/src/components/CreateNewPostForm.jsx

```diff
@@ -28,20 +29,27 @@ const createNewPostForm = () => {
         endTime: '',
     });
     const { user } = useAuth();
+    const { loading, setLoading } = useLoading();
     const nav = useNavigate();
 
     const handlePostSubmit = async (e) => {
         e.preventDefault();
         try {
```

Comment:
Instead of having to add setLoading(true) & false around the newPost calls, you can wrap around the try-catch.
```suggestion
        setLoading(true);
        try {
        ....
        } catch (error) {
            console.error(error);
        }
        setLoading(false);
```